tests/pr_test.go

```diff
@@ -163,6 +163,57 @@ func TestRunUpgradeRoksPattern(t *testing.T) {
 	}
 }
 
+func TestValidateOverrideRoks(t *testing.T) {
+	t.Parallel()
+
+	options := setupOptionsRoksPattern(t, "slz-ovr-vld")
+	options.SkipTestTearDown = true
+	output, err := options.RunTestConsistency()
+
+	override_json := filepath.Join(roksPatternTerraformDir, "override.json")
+	config_json := options.LastTestTerraformOutputs["config"]
+	ignored_keys := []string{"resource_group"}
+
+	//Parse the json into a map
+	var override_json_map map[string]interface{}
+	err1 := json.Unmarshal([]byte(override_json), &override_json_map)
+	fmt.Println(err1)
+
+	config_json_byteArray, ok := config_json.([]byte)
+	if !ok {
+		fmt.Println("Error: Unable to convert config_json to []byte")
+		return
+	}
+
+	var config_map map[string]interface{}
+	err2 := json.Unmarshal([]byte(config_json_byteArray), &config_map)
+	fmt.Println(err2)
+
+	//Delete the ignored keys from the map
+	for _, key := range ignored_keys {
+		delete(override_json_map, key)
+		delete(config_map, key)
+	}
+
+	//Convert the maps back to json string
+	new_override_json_string_value, err3 := json.Marshal(override_json_map)
+	new_config_string_value, err4 := json.Marshal(config_map)
+
+	//Compare both the json strings
+	jsonComp, err3 := common.IsJsonEqual(string(new_override_json_string_value), string(new_config_string_value))
+
+	assert.Nil(t, err, "This should not have errored")
+	assert.NotNil(t, output, "Expected some output")
+	assert.NotNil(t, options.LastTestTerraformOutputs, "Expected some Terraform outputs")
+	assert.Nil(t, err1, "Error occured with the override json")
+	assert.Nil(t, err2, "Error occured with config json")
+	assert.Nil(t, err3, "Error occured with override_json_map")
+	assert.Nil(t, err4, "Error occured with config_map")
+	assert.True(t, jsonComp, err3)
+
+	options.TestTearDown()
```

Comment:
I would consider putting the TestTearDown call in a golang `defer`, right after the RunTestConsistency, so that it will be called no matter how the unit test function exits.

Reply:
Added